File: Christine/RotateArray.java
class Solution {
    public static void rev(int[] arr, int beg, int end) {
        while (beg < end) {
            int temp = arr[beg];
            arr[beg++] = arr[end];
            arr[end--] = temp;
        }
    }

    public static void rotate(int[] nums, int k) {
        int n = nums.length;

        k %= n;

        rev(nums, 0, n - 1);
        rev(nums, 0, k - 1);
        rev(nums, k, n - 1);
    }
}

File: Riddhi/rotatearray.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        k = k % len(nums)
        l = len(nums)
        sub_front = nums[l - k: l]
        sub_end = nums[0:l - k]
        index = 0
        for i in range(len(sub_front)):
            nums[index] = sub_front[index]
            index += 1

        for j in range(len(sub_end)):
            nums[index] = sub_end[j]
            index += 1

        
